beforematch-security-and-privacy-assessment.md

```diff
@@ -0,0 +1,167 @@
+
+### 2.3. How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+
+No new information stored on the device is exposed by this feature.
+
+If the user enters this type of information into their device by using the ScrollToTextFragment or find-in-page features, then this feature may expose it as described in the "Privacy Considerations" section.
+
+### 2.4. How does this specification deal with sensitive information?
+
+The search queries present in the ScrollToTextFragment and find-in-page features may be sensitive information, and are considered in the "Privacy Considerations" section.
+
+### 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?
+
+This feature does not store any new data in the browser.
+
+### 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+No information from the underlying platform is exposed by this feature.
+
+### 2.7. Does this specification allow an origin access to sensors on a user's device
+
+No.
+
+### 2.8. What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+This feature does not expose any information about an origin's state and does not enable sending or receving of any data to or from a network connection.
+
+### 2.9. Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+### 2.10. Does this specification allow an origin to access other devices?
+
+No.
+
+### 2.11. Does this specification allow an origin some measure of control over a user agent's native UI?
+
+This feature allows the page to respond to find-in-page, which is part of the user agent's native UI. However, this feature does not allow control over find-in-page.
+
+### 2.12. What temporary identifiers might this specification create or expose to the web?
+
+None.
+
+### 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?
+
+I'm assuming first-party and third-party contexts here refers to cross origin iframes.
+* In the ElementFragment case, the scrolling, fragments, and beforematch events are fully insulated from each other when they occur in different cross origin iframes.
+* In the ScrollToTextFragment case, text inside of a cross origin iframe will not be searched for or scrolled to. Therefore, no beforematch event will be emitted.
+* In the find-in-page case, content in cross origin iframes can be searched for and scrolled to. I'm not sure if and where we should fire beforematch in this case.
```

Comment:
The browser would coordinate searching across different frames, so the event should only fire within the frame that contains a match

Reply:
Right, that makes perfect sense and doesn't really require any additional implementation details. I'll clarify it.